### ISSUE TITLE
Bug in optimization_tf   create_optimizer 

### ISSUE THREAD

**Issue by JunnYu** [huggingface/transformers#3856]
# 🐛 Bug

## Information

WHEN I am using optimization_tf(create_optimizer):
Problems with learning rate schedule
## To reproduce
```
from transformers.optimization_tf import create_optimizer
import matplotlib.pyplot as plt
%matplotlib inline
opt = create_optimizer(init_lr=5e-5, num_train_steps=100, num_warmup_steps=50)
lr = opt.learning_rate
results = [lr(i).numpy() for i in range(101)]
print(results[49:51])
plt.plot(results)
plt.show()
```
output  [4.9e-05, 2.5e-05, 2.45e-05]
## Expected output
the max lr should be 5e-05
expected output [4.9e-05, 5e-05, 4.9e-05]

```
class WarmUp(tf.keras.optimizers.schedules.LearningRateSchedule):
    def __call__(self, step):
        with tf.name_scope(self.name or "WarmUp") as name:
            # Implements polynomial warmup. i.e., if global_step < warmup_steps, the
            # learning rate will be `global_step/num_warmup_steps * init_lr`.
            global_step_float = tf.cast(step, tf.float32)
            warmup_steps_float = tf.cast(self.warmup_steps, tf.float32)
            warmup_percent_done = global_step_float / warmup_steps_float
            warmup_learning_rate = self.initial_learning_rate * tf.math.pow(warmup_percent_done, self.power)
            return tf.cond(
                global_step_float < warmup_steps_float,
                lambda: warmup_learning_rate,
                lambda: self.decay_schedule_fn(step),
                name=name,
            )
```
change:
lambda: self.decay_schedule_fn(step) =>lambda: self.decay_schedule_fn(step-warmup_steps_float)

```
def create_optimizer(init_lr, num_train_steps, num_warmup_steps):
    """Creates an optimizer with learning rate schedule."""
    # Implements linear decay of the learning rate.
    learning_rate_fn = tf.keras.optimizers.schedules.PolynomialDecay(
        initial_learning_rate=init_lr, decay_steps=num_train_steps, end_learning_rate=0.0
    )
    if num_warmup_steps :
        learning_rate_fn = WarmUp(
            initial_learning_rate=init_lr, decay_schedule_fn=learning_rate_fn, warmup_steps=num_warmup_steps
        )
```
change:
PolynomialDecay  decay_steps  should be  num_train_steps-num_warmup_steps

**Comment by amodaresi** [comment]
I have raised https://github.com/huggingface/transformers/pull/4940, waiting for approval.
